tests/import_export_tests.py

```diff
@@ -0,0 +1,386 @@
+            'dash_with_2_slices',
+            slcs=[get_slc('energy_slc', 'energy_usage', 1002),
+                  get_slc('birth_slc', 'birth_names', 1003)],
+            id=1003)
+        imported_dash_id_3 = models.Dashboard.import_obj(
+            dash_with_2_slices, import_time=1991)
+        imported_dash_3 = get_dash(imported_dash_id_3)
+        self.assertEquals('dash_with_2_slices', imported_dash_3.dashboard_title)
+        self.assertEquals(2, len(imported_dash_3.slices))
+        self.assertEquals(
+            {"remote_id": 1003, "import_time": 1991},
+            json.loads(imported_dash_3.json_metadata))
+        imported_dash_slice_ids_3 = {s.id for s in imported_dash_3.slices}
+
+        # Case 4. Override the dashboard and 2 slices
+        dash_with_2_slices_new = create_dashboard(
+            'dash_with_2_slices_new',
+            slcs=[get_slc('energy_slc', 'energy_usage', 1002),
+                  get_slc('birth_slc', 'birth_names', 1003)],
+            id=1003)
+        imported_dash_id_4 = models.Dashboard.import_obj(
+            dash_with_2_slices_new, import_time=1992)
+        imported_dash_4 = get_dash(imported_dash_id_4)
+        self.assertEquals('dash_with_2_slices_new',
+                          imported_dash_4.dashboard_title)
+        self.assertEquals(2, len(imported_dash_4.slices))
+        self.assertEquals(
+            imported_dash_slice_ids_3, {s.id for s in imported_dash_4.slices})
+        self.assertEquals(
+            {"remote_id": 1003, "import_time": 1992},
+            json.loads(imported_dash_4.json_metadata))
+        # override doesn't change the id
+        self.assertEquals(imported_dash_id_3, imported_dash_id_4)
+
+        # cleanup
+        slc_to_delete = set(imported_dash_2.slices)
+        slc_to_delete.update(imported_dash_4.slices)
+        for slc in slc_to_delete:
+            session.delete(slc)
+
+        session.delete(imported_dash_1)
+        session.delete(imported_dash_2)
+        session.delete(imported_dash_4)
+        session.commit()
+
+    def test_import_tables(self):
+        session = db.session
+
+        def create_table(
+                name, schema='', id=0, cols_names=[], metric_names=[]):
+            json_metadata = {'remote_id': id, 'database_name': 'main'}
+            table = models.SqlaTable(
+                id=id,
+                schema=schema,
+                table_name=name,
+                json_metadata=json.dumps(json_metadata)
+            )
+            for col_name in cols_names:
+                table.columns.append(
+                    models.TableColumn(column_name=col_name))
+            for metric_name in metric_names:
+                table.metrics.append(models.SqlMetric(metric_name=metric_name))
+            return table
+
+        def get_table(table_id):
+            return session.query(models.SqlaTable).filter_by(
+                id=table_id).first()
+
+        # Case 1. Import table with no metrics and columns
```

Comment:
yeah, in this case test cases are build upon each other. I'll try to separate them.